terraform/ecs.tf

```
@@ -0,0 +1,56 @@
+# --------------------------
+# Cluster
+# --------------------------
+resource "aws_ecs_cluster" "tasting_note" {
+  name = "${var.project}-cluster"
+}
+
+# --------------------------
+# task difinition
+# --------------------------
+resource "aws_ecs_task_definition" "rails_task" {
+  family                   = "${var.project}-task"
+  cpu                      = 256
+  memory                   = 512
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+
+  execution_role_arn = aws_iam_role.ecs_task_execution.arn
+
+  container_definitions = templatefile("./task_definitions/rails_container_definitions.json", {
+    project               = var.project
+    image                 = aws_ecr_repository.tasting_note.repository_url
+    aws_ssm_parameter_arn = data.aws_ssm_parameter.rails_master_key.arn
+  })
+}
+
+# --------------------------
+# service
+# --------------------------
+resource "aws_ecs_service" "tasting-note" {
+  name                              = "${var.project}-service"
+  cluster                           = aws_ecs_cluster.tasting_note.arn
+  task_definition                   = aws_ecs_task_definition.rails_task.arn
+  desired_count                     = 2
+  launch_type                       = "FARGATE"
+  platform_version                  = "1.4.0"
+  health_check_grace_period_seconds = 60
+
+  network_configuration {
+    assign_public_ip = false
+    security_groups  = [aws_security_group.ecs_sg.id]
+    subnets          = [for subnet in aws_subnet.ecs_private : subnet.id]
```

ここもSplat Expressionsで書ける。

```suggestion
    subnets          = aws_subnet.ecs_private[*].id
```

---

上記同様です。